torch/distributed/optim/__init__.py

```diff
@@ -24,6 +24,9 @@
 from .named_optimizer import _NamedOptimizer
 from .utils import as_functional_optim
 
+from warnings import warn
+warn("TorchScript support for functional optimizers is"
```

Comment:
@mlazos can you give some explanation of what that change is? I don't see it explained in the PR description?

This looks quite worrying because anyone that is importing torch.distributed.optim will now see this warning. Is your plan to remove this whole folder altogether? There are a lot of things here that have nothing to do with torchscipt or functional optimizers?

Reply:
So every functional optimizer in that folder has the `torch.jit.script` decorator. Since this is the main way those are imported, rather than put the warning in each file I thought here was more optimal. I'll just remove the `torch.jit.script` decorator from each functional optim, not remove the whole folder. 

I can add this to the PR description, sorry I thought the intention was clear from our discussion.

Reply:
But there are many other things than these functional optimizers here?
In any case, I'll let @wconstab / @wanchaol  make the decision on the best way forward here since it is distributed code!